mcp/mcp-cli/src/main/java/software/amazon/smithy/java/mcp/cli/ProcessIoProxy.java

```diff
@@ -0,0 +1,274 @@
+                        if (running.get()) {
+                            LOG.error(errorMessage, e);
+                        }
+                    } finally {
+                        if (closeOutput) {
+                            try {
+                                output.close();
+                            } catch (IOException e) {
+                                LOG.error("Error closing stream", e);
+                            }
+                        }
+                    }
+                });
+    }
+
+    /**
+     * Starts the proxy, launching the subprocess and beginning to forward stdin/stdout.
+     *
+     * @throws RuntimeException If there is an error starting the process
+     */
+    public synchronized void start() {
+        if (process != null && process.isAlive()) {
+            return;
+        }
+        try {
+            process = processBuilder.start();
+            running.getAndSet(true);
+
+            // Thread to forward input to process
+            inputThread = createForwardingThread(
+                    process,
+                    inputStream,
+                    process.getOutputStream(),
+                    "process-stdin-forwarder",
+                    "Error forwarding input to process",
+                    true, // Close the process input stream when done
+                    running);
+
+            // Thread to forward process stdout to output
+            outputThread = createForwardingThread(
+                    process,
+                    process.getInputStream(),
+                    outputStream,
+                    "process-stdout-forwarder",
+                    "Error forwarding process stdout",
+                    false, // Don't close the output stream
+                    running);
+
+            // Thread to forward process stderr to error
+            errorThread = createForwardingThread(
+                    process,
+                    process.getErrorStream(),
+                    errorStream,
+                    "process-stderr-forwarder",
+                    "Error forwarding process stderr",
+                    false, // Don't close the error stream
+                    running);
+
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to start process: " + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Shuts down the proxy, stopping all forwarding and terminating the subprocess.
+     *
+     * @return A CompletableFuture that completes when shutdown is finished
+     */
+    public CompletableFuture<Void> shutdown() {
+        return CompletableFuture.runAsync(() -> {
+            synchronized (this) {
+                if (process != null && process.isAlive()) {
+                    try {
+                        // Interrupt the threads
+                        interruptThread(inputThread);
+                        interruptThread(outputThread);
+                        interruptThread(errorThread);
```

Comment:
Start with this before interrupting the threads so each has a chance to exit cleanly. Otherwise, we should expect to see InterruptedExceptions logged by each of the error handling blocks on each shutdown.